themes/wporg-translate-events-2024/blocks/components/event/attendance-mode.php

```diff
@@ -0,0 +1,14 @@
+<?php namespace Wporg\TranslationEvents\Theme_2024;
+use Wporg\TranslationEvents\Translation_Events;
+
+register_block_type(
+	'wporg-translate-events-2024/components-event-attendance-mode',
```

Comment:
The idea was to have the directory a bit more organized so we can have something like this, see below;
`/blocks/components/event` - holds all the different components of an event (title, date, excerpt, list etc)
`/blocks/components/attendee` - holds all the components of an attendee e.g (Name, profile url, profile picture url etc.) 
Similar to the `/blocks/pages/events` directory.

Reply:
I see what you mean, and I agree it's useful to somehow group related blocks together. However, I think the term "component" when in the context of the block editor is commonly used to refer to React components, and not blocks. So I think that using the term component to refer to a block can be a bit confusing.

For example, if you look at [the core `button` block](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/button), you can see that, internally, [it uses the `RichText` component](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-library/src/button/edit.js#L270C6-L270C14), which comes from [`gutenberg/packages/block-editor/src/components/rich-text/`](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-editor/src/components/rich-text/native/index.native.js#L104). Note that `RichText` is not a block, it's a React component that is meant to be reused by blocks.

Currently, we are not using React components because we haven't had to manipulate how blocks render in the block editor, but if in the future we need to do so, at that point we would have both React components, and block components, which I think would be confusing.

So instead of grouping related blocks by placing them in a `components/foo` directory, we could instead prefix the block with it's logical "thing", e.g. `foo-title`. In this scenario, the blocks directory would look something like:

- `blocks/`
    - `event-title/`
    - `event-date/`
    - ...
    - `attendee-name/`
    - `attendee-profile-url/`
    - ...

These blocks can then be referenced with `wporg-translate-events/event-title` or `wporg-translate-events/attendee-name`.

This is what I was suggesting, sorry I should I have made it clearer.

Reply:
Thank you so much for the detailed response, it's a lot clearer now and thanks for the reference to [gutenberg core](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src)